src/test/java/com/team701/buddymatcher/controllers/users/UserControllerIntegrationTest.java

```diff
@@ -142,6 +142,33 @@ void createAndDeleteUserBuddy() throws Exception {
                 .andExpect(status().isOk());
     }
 
+    @Test
+    void reportUser() throws Exception {
+        // Test that a POST request for user 2 to report user 1 is OK
+        mvc.perform(post("/api/users/report/{id}", 1)
+                        .sessionAttrs(Collections.singletonMap("UserId", 2))
+                        .queryParam("reportInfo", "This is a test."))
+                .andExpect(status().isOk());
+
+        // Test that a POST request for user 2 to report user 1 again is OK
```

Comment:
Does it make much sense for the same user to be able to report against the same person multiple times? I think they should be only allowed to report another user once otherwise, you could have someone spamming reports on someone.

Reply:
The idea is that the table will be for admin to read and resolve reports by users. For now a user can make as many reports as they like on any other user - there is no exact format for reports aside for it being a string, nor is anything implemented at the frontend for this. A user may spam, or a user may report more than one found offence by any other user.

Reply:
Alright, if you think it's fine to keep it this way for now then all goods. The rest looks fine, so I'll approve it now, feel free to merge it in.